### ISSUE TITLE
Image Sharing `.sharefileprovider` fails

### ISSUE THREAD

**Issue by Dayruke** [cengiz-pz/godot-android-share-plugin#7]
### Environment

Linux Mint 20.3
Godot 4.3.stable (double precision)
Share Plugin v3.0
Android SDK 34
JDK 17.0.12+7

### Error

Sharing images fails. Sharing text works fine. 

I have gotten image share to work by changing my Export Settings:

`Project > Export > Android (Preset) > Package > Unique Name`

to the value 

`org.godotengine.plugin.android.share`

![image](https://github.com/user-attachments/assets/6f1c275f-b7fd-4555-aef2-9932101623c5)

If it's something like `com.my.packagename`, I'll get this error (from adb)

```
09-19 14:48:49.251 32009 32062 E godot::SharePlugin: The selected file can't be shared: /data/data/com.my.packagename/files/tmp_share_img_path.png
09-19 14:48:49.251 32009 32062 E godot::SharePlugin: java.lang.IllegalArgumentException: Couldn't find meta-data for provider with authority com.my.packagename.sharefileprovider
```
The important bit:

```
Couldn't find meta-data for provider with authority com.my.packagename.sharefileprovider
```

I don't imagine this is the intended behavior. I'd like to have the project be under some name other than the name of your plugin (great, though it is).

I think the problem may be related to `ShareExportPlugin.gd` where you do this?

```
android:authorities="%s.sharefileprovider"
<...>
func _get_android_manifest_application_element_contents(platform: EditorExportPlatform, debug: bool) -> String:
		return PROVIDER_TAG % [PLUGIN_PACKAGE, get_option("package/unique_name")]
```

**Comment by cengiz-pz** [comment]
Hi @Dayruke, could you please attach a minimal project that reproduces this error?  (Just your code, plugin not necessary)

**Comment by Dayruke** [comment]
Thanks for your response. Your request indicated that I was probably unique with this issue, so I kept at it.

*I wish I could tell you what resolved the issue!*

For posterity: I tried making my project's package name all lowercase letters. After that, sharing an image worked fine, but I'd be surprised if *that* was the solution.

I was recently tinkering with keystores, too. So maybe something changed in the export process or the environment it created on my target phone that allowed the metadata to be found (which was apparently missing, according to the error message in my first post)?

I'll close this issue. Thanks, again and well done on these sharing plugins. I'm grateful to have this and the iOS one for my project.

**Comment by Dayruke** [comment]
I'll also add that I was using 'Remote Debug' in the Godot editor when the issue first cropped up.

I have a vague idea that the Android OS treats the application differently if it's installed the usual way (clicking an `.apk`) vs remote debug.

**Comment by hero-clashes** [comment]
can confirm the same bug exists, the naming of the package needs to be set for files sharing to work

**Comment by cengiz-pz** [comment]
Yes, having a placeholder in the package name will not work.  It needs to be replaced with a real package name.

**Comment by sebageek** [comment]
Can confirm as well. The default package name `com.example.$genname` in the android export template will not work. Changing the `package/unique_name` to something without  placeholder/variables made it work for me.
